### PR TITLE
Logger file  issue resolved

### PR DESCRIPTION
**Pull request by amenasolvd**

### CODE REVIEW

File: src/main/Library.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            for (Book i : bookList.getAll()) {
+                if (book.getTitle().contains(searchTitle)) {
+                    LOGGER.info("Search result:  " + book);
+                }
+            }
+            LOGGER.info("no Book found");
+        } catch (Exception e) {
+            LOGGER.info("Give valid search input");
+        }
+    }
```
Comment (ankochergin) [inline src/main/Library.java]: please check business logic in this method. method doesn't work well now

File: logs/xmlfilelog.log [hunk 2]
```diff
@@ -0,0 +1,20 @@
+2023-11-29 19:38:06,114 INFO m.Library [main] member added
```
Comment (ankochergin) [inline logs/xmlfilelog.log]: remove log file from the git

File: src/linkedlist/CustomLinkedList.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    int getLength() {
+        return this.length;
+    }
+
+    boolean isEmpty() {
+        // Checking if next node points to null
+        return head == null;
+    }
```
Comment (ankochergin) [inline src/linkedlist/CustomLinkedList.java]: access modifiers

File: src/library/Library.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                }
+            }
+            if (!allFound) {
+                LOGGER.info("no Book found");
+            }
+
+        } catch (Exception e) {
+            LOGGER.info("Give valid search input");
+        }
+    }
```
Comment (ankochergin) [inline src/library/Library.java]: Still method doesn't make sense. Method should search book by title. 
1. Lets use bookTitle as an argument of the method.
2. result of searchBook I believe is a book. So your method should return book

File: src/library/Library.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+            DateTimeFormatter dtf = DateTimeFormatter.ofPattern(newsDate);
+            LocalDate date = LocalDate.parse(newsDate);
+            if (date.isAfter(localDate)) {
+                throw new InvalidInputException("Date is invalid");
+            }
+            newspaperList.add(newspaper);
+        } catch (InvalidInputException e) {
+            LOGGER.error("Enter Date in correct format ");
+        }
+    }
```
Comment (ankochergin) [inline src/library/Library.java]: please check this method too. addNewspaper method has newspaper argument, you can just add it to the list of your newspapers. I think you don't need newspaperTitle and date if it's already in your newspaper object.

File: src/library/Library.java [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+    @Override
+    public boolean reissue(Member member, Book book) throws ReissueNotValidException {
+        try {
+            int reissueCount = 0;
+            if (reissueCount < 1) {
+                this.issue(member, book);
+                reissueCount++;
+                return true;
+            }
```
Comment (ankochergin) [inline src/library/Library.java]: please simplify this code. I'm not sure that you need `if` and `reissueCount`

File: src/library/Library.java (hunk 4 above)
Comment (amenasolvd) [reply src/library/Library.java]: then, I need to move scanner outside of method

File: src/library/Library.java (hunk 5 above)
Comment (amenasolvd) [reply src/library/Library.java]: you mean, I won't need here scanner ?

File: src/library/Library.java [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+            for (Book i : bookList.getAll()) {
+                boolean isFound = i.getTitle().contains(bookTitle);
+                if (isFound) {
+                    List<Book> bookFound = new ArrayList<>();
+                    bookFound.add(i);
+                    return bookFound;
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```
Comment (ankochergin) [inline src/library/Library.java]: what does throw a runtime exception here?

File: src/linkedlist/CustomLinkedList.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    public int getLength() {
+        return this.length;
+    }
+
+    public boolean isEmpty() {
+        return head == null;
+    }
+
+    @Override public void add(T data) {
+        Node<T> newNode = new Node<>(data);
```
Comment (ankochergin) [inline src/linkedlist/CustomLinkedList.java]:
```suggestion
    @Override 
    public void add(T data) {
        Node<T> newNode = new Node<>(data);
```

Comment (ankochergin) [reply src/linkedlist/CustomLinkedList.java]: same in other methods here

**Pull request merged**